src/Disks/IO/ThreadPoolRemoteFSReader.cpp

```diff
@@ -68,14 +73,42 @@ ThreadPoolRemoteFSReader::ThreadPoolRemoteFSReader(size_t pool_size, size_t queu
 
 std::future<IAsynchronousReader::Result> ThreadPoolRemoteFSReader::submit(Request request)
 {
+    auto * fd = assert_cast<RemoteFSFileDescriptor *>(request.descriptor.get());
+    auto & reader = fd->getReader();
+
+    {
+        ProfileEventTimeIncrement<Microseconds> elapsed(ProfileEvents::ThreadpoolReaderPrepareMicroseconds);
+        /// `seek` have to be done before checking `isContentCached`, and `set` have to be done prior to `seek`
+        reader.set(request.buf, request.size);
+        reader.seek(request.offset, SEEK_SET);
+    }
+
+    if (reader.isContentCached(request.offset, request.size))
```

Comment:
Would be good to have `isContentCached` under some metric as well, because it can call `initialize` method and it will be good to be able to know if this method is not as cheap as we think.